### PR TITLE
Add tvheadend:4.0.4

### PR DESCRIPTION
**Pull request by simonvandel**
Since this is my first commit for nixpkgs, is anything missing? Any things done non-idiomatic?

### CODE REVIEW

File: nixos/modules/services/networking/tvheadend.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+with lib;
+
+let cfg     = config.services.tvheadend;
+    pidFile = "${config.users.extraUsers.tvheadend.home}/tvheadend.pid";
+in
+
+{
+  options = {
+    services.tvheadend = {
+      enable = mkOption {
```
Comment (jagajaga) [inline nixos/modules/services/networking/tvheadend.nix]: Use `mkEnableOption`.

File: nixos/modules/services/networking/tvheadend.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        type = types.bool;
+        default = false;
+        description = "If enabled, start Tvheadend";
+      };
+
+      httpPort = mkOption {
+        type        = types.int;
+        default     = 9981;
+        description = "Port to bind HTTP to";
```
Comment (jagajaga) [inline nixos/modules/services/networking/tvheadend.nix]: Add dots at the ends of the description.

File: pkgs/servers/tvheadend/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  enableParallelBuilding = true;
+
+  buildInputs = [ avahi dbus git gzip libav libiconv openssl pkgconfig python which zlib];
+
+  preConfigure = "patchShebangs ./configure";
+
+  meta = {
+    description = "TV steaming server";
+    longDescription = ''
+	Tvheadend is a TV streaming server and recorder for Linux, FreeBSD and Android supporting DVB-S, DVB-S2, DVB-C, DVB-T, ATSC, IPTV, SAT>IP and HDHomeRun as input sources.
```
Comment (jagajaga) [inline pkgs/servers/tvheadend/default.nix]: Change to
```
Tvheadend is a TV streaming server and recorder for Linux, FreeBSD and Android
supporting DVB-S, DVB-S2, DVB-C, DVB-T, ATSC, IPTV, SAT>IP and HDHomeRun as input sources.
Tvheadend offers the HTTP (VLC, MPlayer), HTSP (Kodi, Movian) and SAT>IP streaming.
``` :smile:

**Comment by jagajaga** [comment]
Rename commit as it said in `CONTRIBUTING.md`.

File: pkgs/servers/tvheadend/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  name = "${pkgName}-${version}";
+
+  src = fetchurl {
+    url = "https://github.com/tvheadend/tvheadend/archive/v${version}.tar.gz";
+    sha256 = "acc5c852bccb32d6a281f523e78a1cceb4d41987fe015aba3f66e1898b02c168";
+  };
+
+  enableParallelBuilding = true;
+
+  buildInputs = [ avahi dbus git gzip libav libiconv openssl pkgconfig python which zlib];
```
Comment (jagajaga) [inline pkgs/servers/tvheadend/default.nix]: Space before `]`.

File: pkgs/servers/tvheadend/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{avahi, dbus, fetchurl, git, gzip, libav, libiconv, openssl, pkgconfig, python, stdenv, which, zlib}:
```
Comment (jagajaga) [inline pkgs/servers/tvheadend/default.nix]: Space after `{` and before `}`.

File: nixos/modules/misc/ids.nix [hunk 6]
```diff
@@ -221,6 +221,7 @@
       skydns = 197;
       ripple-rest = 198;
       nix-serve = 199;
+      tvheadend = 200;
```
Comment (jagajaga) [inline nixos/modules/misc/ids.nix]: Don't you need extragroup? 
Anyway create it and comment with `#` if you don't need.

Comment (simonvandel) [inline nixos/modules/misc/ids.nix]: I don't think the extra group is needed, so created as unused in latest commit.

**Pull request merged**